### DISCUSSION TITLE
sitemap.xml file error

### DISCUSSION THREAD

**Discussion by hassanah07** [vercel/community#3819] [Help]
### Page to Investigate

https://nextjs.org/docs/app/api-reference/file-conventions/metadata/sitemap#sitemap-return-type

### Steps to Reproduce

```
import React from "react";

const Sitemap = async () => {
  const baseUrl = process.env.FRONTEND_LINK;
  const response = await fetch(
    `${process.env.NEXT_PUBLIC_HOST}/api/blogpost/sitemap`,
    {
      cache: "no-cache",
      method: "GET",
      headers: {
        "Content-Type": "application/json"
      }
    }
  );

  const data = await response.json();

  const postUrls = Object.keys(data).map((item) => ({
    url: `${baseUrl}/blog/${data[item].slug}`,
    lastModified: data[item].updatedAt
  }));
  return [
    { url: baseUrl, lastModified: new Date() },
    { url: `${baseUrl}/contact`, lastModified: new Date() },
    { url: `${baseUrl}/login`, lastModified: new Date() },
    { url: `${baseUrl}/pricing`, lastModified: new Date() },
    { url: `${baseUrl}/talk`, lastModified: new Date() },

    ...postUrls
  ];
};

export default Sitemap;
```
=> In development: Everything is ok
=>In production: New Data is not coming to Sitemap.
=> It is not a good practice that After each new post I have to access my vps and need to run command npm run build to get new data in sitemap.xml page. Please solve this issue as soon as possible.
=> I am using latest version of NEXTJS and NodeJS

**Comment by amyegan** [comment]
Hey @hassanah07. I need more context to help me understand the situation. How does `/api/blogpost/sitemap` get its list of posts? Is your site using ISR or some other strategy to create new posts without creating a new deployment? If you share your public repo or a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), that will give me a better view of what is going wrong.
